core/jobs/src/main/java/com/cognifide/aet/job/common/modifiers/hide/HideModifier.java

```diff
@@ -92,17 +81,21 @@ public CollectorStepResult hideElement(WebDriver driver, String xpath) throws Pr
       } else {
         script = DISPLAY_NONE_SCRIPT;
       }
-      List<WebElement> webElements = driver.findElements(By.xpath(xpath));
+
+      By elementLocator = getLocator();
+      waitForElementToBePresent(webDriver, elementLocator);
+
+      List<WebElement> webElements = webDriver.findElements(elementLocator);
       for (WebElement element : webElements) {
-        ((JavascriptExecutor) driver).executeScript(script, element);
+        ((JavascriptExecutor) webDriver).executeScript(script, element);
       }
       result = CollectorStepResult.newModifierResult();
-    } catch (NoSuchElementException e) {
-      final String message = String.format("Error while hiding element '%s'. %s", xpath, e.getMessage());
+    } catch (WebDriverException e) {
```

Comment:
We could have two methods: one for finding element and the other for hiding it.
This way we could have two separate try-catch block.
(nice to have)